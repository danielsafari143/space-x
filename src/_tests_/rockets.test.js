import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Items from './data';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import axios from 'axios';
import Rocket from '../components/Rockets';

jest.mock('axios');

const reducer = (state = {
  rockets: { rocket: Items },
}) => state;

const store = configureStore({ reducer });

describe('should test the rocket component', () => {
  it('should test the snapshot of the rocket component', () => {
    axios.get.mockResolvedValue({ data: Items });
    const navb = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <Rocket />
        </Provider>
      </MemoryRouter>,
    ).toJSON();
    expect(navb).toMatchSnapshot();
  });
});

