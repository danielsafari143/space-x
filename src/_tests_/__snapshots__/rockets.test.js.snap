// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test the rocket component should test the snapshot of the rocket component 1`] = `
<div
  className="rocket"
>
  <ul
    className="rocket-list"
  >
    <li
      className="rocket-item"
    >
      <img
        alt="Falcon 1"
        className="rocket-img"
        src={
          Array [
            "https://imgur.com/DaCfMsj.jpg",
            "https://imgur.com/azYafd8.jpg",
          ]
        }
      />
      <div
        className="rocket-container"
      >
        <p
          className="rocket-paragraph-first"
        >
          Falcon 1
        </p>
        <p
          className="rocket-paragraph-first"
        >
           The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.
        </p>
        <button
          className="rocket-btn reserved-rocket"
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </li>
    <li
      className="rocket-item"
    >
      <img
        alt="Falcon 9"
        className="rocket-img"
        src={
          Array [
            "https://farm1.staticflickr.com/929/28787338307_3453a11a77_b.jpg",
            "https://farm4.staticflickr.com/3955/32915197674_eee74d81bb_b.jpg",
            "https://farm1.staticflickr.com/293/32312415025_6841e30bf1_b.jpg",
            "https://farm1.staticflickr.com/623/23660653516_5b6cb301d1_b.jpg",
            "https://farm6.staticflickr.com/5518/31579784413_d853331601_b.jpg",
            "https://farm1.staticflickr.com/745/32394687645_a9c54a34ef_b.jpg",
          ]
        }
      />
      <div
        className="rocket-container"
      >
        <p
          className="rocket-paragraph-first"
        >
          Falcon 9
        </p>
        <p
          className="rocket-paragraph-first"
        >
           Falcon 9 is a two-stage rocket designed and manufactured by SpaceX for the reliable and safe transport of satellites and the Dragon spacecraft into orbit.
        </p>
        <button
          className="rocket-btn reserved-rocket"
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </li>
    <li
      className="rocket-item"
    >
      <img
        alt="Falcon Heavy"
        className="rocket-img"
        src={
          Array [
            "https://farm5.staticflickr.com/4599/38583829295_581f34dd84_b.jpg",
            "https://farm5.staticflickr.com/4645/38583830575_3f0f7215e6_b.jpg",
            "https://farm5.staticflickr.com/4696/40126460511_b15bf84c85_b.jpg",
            "https://farm5.staticflickr.com/4711/40126461411_aabc643fd8_b.jpg",
          ]
        }
      />
      <div
        className="rocket-container"
      >
        <p
          className="rocket-paragraph-first"
        >
          Falcon Heavy
        </p>
        <p
          className="rocket-paragraph-first"
        >
           With the ability to lift into orbit over 54 metric tons (119,000 lb)--a mass equivalent to a 737 jetliner loaded with passengers, crew, luggage and fuel--Falcon Heavy can lift more than twice the payload of the next closest operational vehicle, the Delta IV Heavy, at one-third the cost.
        </p>
        <button
          className="rocket-btn reserved-rocket"
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </li>
    <li
      className="rocket-item"
    >
      <img
        alt="Starship"
        className="rocket-img"
        src={
          Array [
            "https://live.staticflickr.com/65535/48954138962_ee541e6755_b.jpg",
            "https://live.staticflickr.com/65535/48953946911_e60c5bcc5c_b.jpg",
            "https://live.staticflickr.com/65535/48954138922_9c42173f08_b.jpg",
            "https://live.staticflickr.com/65535/48953947006_313f01ec93_b.jpg",
          ]
        }
      />
      <div
        className="rocket-container"
      >
        <p
          className="rocket-paragraph-first"
        >
          Starship
        </p>
        <p
          className="rocket-paragraph-first"
        >
           Starship and Super Heavy Rocket represent a fully reusable transportation system designed to service all Earth orbit needs as well as the Moon and Mars. This two-stage vehicle — composed of the Super Heavy rocket (booster) and Starship (ship) — will eventually replace Falcon 9, Falcon Heavy and Dragon.
        </p>
        <button
          className="rocket-btn reserved-rocket"
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </li>
  </ul>
</div>
`;
